### PR TITLE
Fix Gid implementation

### PR DESCRIPTION
**Pull request by Nugine**
related: https://github.com/datenlord/async-rdma/issues/54#issuecomment-1104750536

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -67,18 +68,26 @@ impl Context {
...
-        }
+
+        let gid = {
+            let mut gid = MaybeUninit::<ibv_gid>::uninit();
+            let gid_index = gid_index.cast();
+            // SAFETY: ffi
+            let errno =
+                unsafe { ibv_query_gid(inner_ctx.as_ptr(), port_num, gid_index, gid.as_mut_ptr()) };
+            if errno != 0_i32 {
+                assert_ne!(errno, i32::MIN);
```
Comment (rogercloud) [inline src/context.rs:79]: What's purpose of this assertion?

File: src/gid.rs [hunk 2]
```diff
@@ -1,63 +1,96 @@
...
-#[derive(
-    serde::Serialize, serde::Deserialize, Default, Copy, Clone, Debug, Eq, PartialEq, Hash,
-)]
+#[derive(Clone, Copy)]
 #[repr(transparent)]
-pub(crate) struct Gid {
-    /// Gid raw data
-    raw: [u8; 16],
-}
+pub(crate) struct Gid(ibv_gid);
```
Comment (rogercloud) [inline src/gid.rs:9]: Add a naive test to check alignment?

File: src/context.rs (hunk 1 above)
Comment (Nugine) [reply src/context.rs:79]: <https://doc.rust-lang.org/stable/std/primitive.i32.html#method.wrapping_neg>
It seems unnecessary. I will remove it.

Comment (rogercloud) [reply src/context.rs]: The possible return value is 0 or -1, we won't i32::MIN here. Ref: https://www.rdmamojo.com/2012/08/02/ibv_query_gid/

**Review by rogercloud — APPROVED** [review]

**Pull request merged**